frontend/src/components/ScratchList.tsx

```diff
@@ -82,9 +82,8 @@ export function getMatchPercentString(scratch: api.TerseScratch) {
 export function ScratchItem({ scratch, children } : { scratch: api.TerseScratch, children?: ReactNode }) {
     const compilersTranslation = useTranslation("compilers")
     const compilerName = compilersTranslation.t(scratch.compiler as any)
-    const serverPresets = api.usePlatforms()[scratch.platform].presets
     const matchPercentString = getMatchPercentString(scratch)
-    const preset = serverPresets.find(p => p.id === scratch.preset)
+    const preset = api.usePreset(scratch.preset)
     const presetName = preset?.name
 
     const presetOrCompiler = presetName ?
```

Comment:
Consider changing L92 to use a loading skeleton element in place of the preset name - it looks a bit iffy to flash from compiler name to preset name once usePreset finishes loading

Reply:
Actually, there will be a flash regardless because of the width and there being text to the right. Best thing to do is to view transition this I think, regardless of whether its a skeleton or text.

Shouldn't need to block this PR though, we should make an issue out of it if it's nontrivial

Reply:
Do you mind creating an issue so it's not forgotten? I don't know what "view transition this" means :(